Sources/View/Widget/DateWidget.swift

```diff
@@ -0,0 +1,98 @@
+//
+// Copyright 2018 - Present Hyperwallet
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software
+// and associated documentation files (the "Software"), to deal in the Software without restriction,
+// including without limitation the rights to use, copy, modify, merge, publish, distribute,
+// sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or
+// substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import HyperwalletSDK
+
+/// Represents the date input widget.
+final class DateWidget: TextWidget {
+    private var datePicker: UIDatePicker!
+    private var toolbar: UIToolbar!
+    private static var dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd"
+        return formatter
+    }()
+
+    required init(field: HyperwalletField) {
+        super.init(field: field)
+        setupLayout(field: field)
+    }
+
+    required init(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+
+    override func setupLayout(field: HyperwalletField) {
+        super.setupLayout(field: field)
+        setupDatePicker()
+        setupToolbar()
+        setupTextField()
+    }
+
+    private func setupDatePicker() {
+        datePicker = UIDatePicker(frame: .zero)
+        datePicker.datePickerMode = .date
+        datePicker.addTarget(self, action: #selector(updateTextField), for: .valueChanged)
+
+        datePicker.accessibilityIdentifier = "DateWidgetPicker"
+    }
+
+    private func setupToolbar() {
```

Comment:
This code is duplicate. It's in PickerView as well. Maybe create an extension of UIToolBar and add a func to setUpToolBar(...) passing the selector action.

Reply:
Also aftre moving common logic to extension, check if PickerView class is really required or not. If not then delete it and modify the logic of MonthYearPicker to handle the done tap